### DISCUSSION TITLE
Shorthand for .difference(DateTime.now())

### DISCUSSION THREAD

**Discussion by ChristianKleineidam** [jogboms/time.dart#54] [General]
I dislike writing `.difference(DateTime.now())`. On the homepage of the package I didn't find a shorthand for that expression in this package? Does one exist (and maybe might be made easier to find)?

**Comment by jogboms** [comment]
Hey @ChristianKleineidam 

As operator overload is not yet supported, I only included support for subtracting a `Duration` as opposed a `DateTime`.

**Comment by ChristianKleineidam** [comment]
Operator overloading isn't directly supported but the functionality can be expressed with `is`. Why not do:

```
DateTime operator -(Comparable otherTime) {
  if (otherTime is Duration){
    return subtract(otherTime);
  } else if (otherTime is DateTime){
    return difference(otherTime)
  } else {
    throw IllegalArgumentException("${otherTime.toString()} is neither Duration() nor DateTime()");
  }
}
```

**Reply by FMorschel** [reply]
About this, you would have two return types, the `subtract` returns a `DateTime` and the `difference` returns a `Duration`.

**Comment by shinriyo** [comment]
@jogboms
Are you busy?

**Comment by jogboms** [comment]
Hey @shinriyo 
I was but took this weekend to review the outstanding issues and PRs.
I found this one not currently a possibility as we still do not have language support for union types which is what the PR proposes. I believe it needs for refinement.
BTW I dropped my comments on the PR.
